File: src/main/java/rjojjr/com/github/taskslock/models/TaskLock.java
package rjojjr.com.github.taskslock.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class TaskLock {
    private String taskName;
    private String contextId;
    private Boolean isLocked;
    private Date lockedAt;
    /**
     * Number of minutes after {@link #lockedAt} at which the lock is considered
     * expired and may be acquired by another requester. {@code null} means the
     * lock never expires.
     */
    private Long timeoutMinutes;
    private Runnable release;
}

File: client-modules/python/src/taskslock/client/models/task_lock.py
from typing import Callable


class TaskLock:
    def __init__(self, api_response: dict, release: Callable):
        self.task_name = api_response['taskName']
        self.context_id = api_response['contextId']
        self.is_locked = api_response['isLockAcquired']
        self.release = release
